### PR TITLE
Camera angle and fixes

### PR DESCRIPTION
**Pull request by JamesWP**

### CODE REVIEW

File: src/JavascriptVM/JavascriptVM.tsx [hunk 1]
```diff
@@ -170,7 +170,59 @@ export const VMProvider: FunctionComponent = ({ children }) => {
     return () => {
       console.log("Destroying VMContext.Provider");
     };
-  });
+  }, [canvasRef]);
+
+  function setSimulatorCameraMode(
+    simulator: Sim3D,
+    robotRef: Handles.RobotHandle,
+    cameraMode: CameraView
```
Comment (ido-ran) [inline src/JavascriptVM/JavascriptVM.tsx:178]: There is a mismatch here with cameraMode and CameraView.
I think we should rename CameraView into CameraMode.

**Comment by ido-ran** [comment]
The new top-down values of x:0 y:7 z:3 produce a view which seems a bit far but does present 3D view.
image

What do you think about using x:0 y:4 z:0 for a complete top-down view
image

File: src/JavascriptVM/JavascriptVM.tsx [hunk 2]
```diff
@@ -39,6 +39,7 @@ import {
   challengeSlice,
   ChallengeStatus,
 } from "../RobotSimulator/Arenas/challengeSlice";
+import { SimBasicSensor } from "@fruk/simulator-core/dist/engine/objects/robot/sensors/SimBasicSensor";
```
Comment (ido-ran) [inline src/JavascriptVM/JavascriptVM.tsx:42]: Why was that import added?

**Review by ido-ran — COMMENTED** [review]
This fix is for https://github.com/FRUK-Simulator/Simulator/issues/187, right?

It is more bird-eye but do you want to make it completely top-down?

File: src/JavascriptVM/JavascriptVM.tsx (hunk 1 above)
Comment (JamesWP) [reply src/JavascriptVM/JavascriptVM.tsx:178]: The parameters here are slightly confusing! but i wont try to fix that in this PR

File: src/JavascriptVM/JavascriptVM.tsx (hunk 2 above)
Comment (JamesWP) [reply src/JavascriptVM/JavascriptVM.tsx]: That was a mistake, i think VsCode has a habbit of adding these when they're not needed! :)
